vscode/src/issueDiagnosingHelper.ts

```diff
@@ -0,0 +1,141 @@
+import * as vscode from "vscode";
+
+import { Workspace } from "./workspace";
+
+export async function diagnoseIssues(workspace: Workspace | undefined) {
+  if (!workspace) {
+    await vscode.window.showErrorMessage("No active Ruby workspace found.");
+    return;
+  }
+
+  const diagnosticInfo = await collectDiagnosticInfo(workspace);
+  const panel = vscode.window.createWebviewPanel(
+    "rubyLspDiagnostics",
+    "Ruby LSP Diagnostics",
+    vscode.ViewColumn.One,
+    { enableScripts: true },
+  );
+
+  panel.webview.html = generateDiagnosticsHtml(diagnosticInfo);
+}
+
+async function collectDiagnosticInfo(
+  workspace: Workspace,
+): Promise<Record<string, string | string[]>> {
+  const vscodeVersion = vscode.version;
+  const rubyLspExtension = vscode.extensions.getExtension("Shopify.ruby-lsp")!;
+  const rubyLspExtensionVersion = rubyLspExtension.packageJSON.version;
+  const rubyLspVersion = workspace.lspClient?.serverVersion ?? "Unknown";
+  const rubyLspAddons =
+    workspace.lspClient?.addons?.map((addon) => addon.name) ?? [];
+  const extensions = await getPublicExtensions();
+
+  return {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "VS Code Version": vscodeVersion,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "Ruby LSP Extension Version": rubyLspExtensionVersion,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "Ruby LSP Server Version": rubyLspVersion,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "Ruby LSP Addons": rubyLspAddons,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "Installed Extensions": extensions,
```

Comment:
We can use the block disable syntax to avoid repeating the comments.
```suggestion
    /* eslint-disable @typescript-eslint/naming-convention */
    "VS Code Version": vscodeVersion,
    "Ruby LSP Extension Version": rubyLspExtensionVersion,
    "Ruby LSP Server Version": rubyLspVersion,
    "Ruby LSP Addons": rubyLspAddons,
    "Installed Extensions": extensions,
    /* eslint-enable @typescript-eslint/naming-convention */
```